apps/mobile/src/components/Gallery/CollectionVirtualizedRow.tsx

```diff
@@ -0,0 +1,57 @@
+import clsx from 'clsx';
+import { useWindowDimensions, View } from 'react-native';
+
+import { WhitespaceBlock } from '~/components/Gallery/createVirtualizedCollectionRows';
+import { GalleryTokenPreview } from '~/components/Gallery/GalleryTokenPreview';
+import { GalleryTokenPreviewFragment$key } from '~/generated/GalleryTokenPreviewFragment.graphql';
+
+type CollectionRowProps = {
+  isFirst: boolean;
+  isLast: boolean;
+  columns: number;
+  items: Array<WhitespaceBlock | GalleryTokenPreviewFragment$key>;
+};
+
+export function CollectionRow({ columns, items, isFirst, isLast }: CollectionRowProps) {
+  const horizontalRowPadding = 16;
+  const inBetweenColumnPadding = 8;
+
+  const screenDimensions = useWindowDimensions();
+  const totalSpaceForTokens =
+    screenDimensions.width - horizontalRowPadding * 2 - inBetweenColumnPadding * (columns - 1);
+
+  const widthPerToken = totalSpaceForTokens / columns;
+
+  return (
+    <View
+      style={{ paddingHorizontal: horizontalRowPadding }}
+      className={clsx('flex w-full flex-row', {
+        'pt-4': isFirst, // First row should be space from header
+        'pt-2': !isFirst, // All the other rows have a bit of space between them
+        'pb-12': isLast, // Last one needs space between itself and the next collection
+      })}
+    >
+      {Array.from({ length: columns }).map((_, index) => {
+        const item = items[index] ?? { whitespace: 'whitespace' };
```

Comment:
I think it's a bit silly that the collection layout parsing doesn't give me 3 items back if the row has 3 columns but that's a problem for another day.